packages/twenty-server/src/engine/core-modules/billing/stripe/stripe.service.ts

```diff
@@ -119,4 +140,29 @@ export class StripeService {
       },
     );
   }
+
+  formatProductPrices(prices: Stripe.Price[]) {
+    const result: Record<string, ProductPriceEntity> = {};
```

Comment:
@charlesBochet 
other way to write it with functional, easier to read imho but a bit less efficient (but could be equivalent if we merge the filter logic in the reduce for example) so up to you 😛 
```typescript
formatProductPrices(prices: Stripe.Price[]): ProductPriceEntity[] {
  return prices
    .filter(item => item.recurring?.interval && item.unit_amount)
    .reduce<Record<string, ProductPriceEntity>>((acc, item) => {
      const interval = item.recurring!.interval;
      
      if (!acc[interval] || item.created > acc[interval].created) {
        acc[interval] = {
          unitAmount: item.unit_amount!,
          recurringInterval: interval,
          created: item.created,
          stripePriceId: item.id,
        };
      }
      
      return acc;
    }, {})
    .values()
    .sort((a, b) => a.unitAmount - b.unitAmount);
}
```